Content.Server/Mech/Systems/MechSystem.cs

```diff
@@ -311,6 +314,10 @@ public override void UpdateUserInterface(EntityUid uid, MechComponent? component
             EquipmentStates = ev.States
         };
         _ui.SetUiState(uid, MechUiKey.Key, state);
+
+        TryComp<BatteryComponent>(component.BatterySlot.ContainedEntity, out var batteryComp);
+
+        _ui.ServerSendUiMessage(uid, MechUiKey.Key, new MechEnergyMessage(batteryComp?.CurrentCharge ?? 0f, batteryComp?.MaxCharge ?? 0f));
```

Comment:
This is the only time `MechEnergyMessage` is instantiated, so why not make the constructor more particular to the usage?
eg.
```csharp
new MechEnergyMessage(batteryComp)
```
and then move the null handling into its implementation?